packages/firestore/test/unit/local/indexeddb_persistence.test.ts

```diff
@@ -712,6 +712,109 @@ describe('IndexedDbSchema: createOrUpgradeDb', () => {
+    // This test creates a database with schema version 8 that has a few
+    // remote documents, adds an index and then reads new documents back
+    // via that index.
+
+    const existingDocPaths = [
+      'coll1/doc1',
+      'coll1/doc2',
+      'coll2/doc1',
+      'coll2/doc2'
+    ];
+    const newDocPaths = [
+      'coll1/doc3',
+      'coll1/doc4',
+      'coll2/doc3',
+      'coll2/doc4'
+    ];
+
+    await withDb(8, db => {
+      const sdb = new SimpleDb(db);
+      return sdb.runTransaction('readwrite', V8_STORES, txn => {
+        const remoteDocumentStore = txn.store<
+          DbRemoteDocumentKey,
+          DbRemoteDocument
+        >(DbRemoteDocument.store);
+
+        // Write the remote document entries.
+        return PersistencePromise.forEach(existingDocPaths, (path: string) => {
+          const remoteDoc = doc(path, /*version=*/ 1, { data: 1 });
+
+          const dbRemoteDoc = TEST_SERIALIZER.toDbRemoteDocument(
+            remoteDoc,
+            remoteDoc.version
+          );
+          // Mimic the old serializer and delete previously unset values
+          delete dbRemoteDoc.readTime;
+          delete dbRemoteDoc.parentPath;
+
+          return remoteDocumentStore.put(
+            remoteDoc.key.path.toArray(),
+            dbRemoteDoc
+          );
+        });
+      });
+    });
+
+    // Migrate to v8 and verify that new documents are indexed.
+    await withDb(9, db => {
+      const sdb = new SimpleDb(db);
+      return sdb.runTransaction('readwrite', V8_STORES, txn => {
+        const remoteDocumentStore = txn.store<
+          DbRemoteDocumentKey,
+          DbRemoteDocument
+        >(DbRemoteDocument.store);
+
+        // Verify the existing remote document entries.
+        return PersistencePromise.forEach(existingDocPaths, (path: string) => {
+          const remoteDoc = doc(path, /*version=*/ 1, { data: 1 });
+
+          return remoteDocumentStore
+            .get(remoteDoc.key.path.toArray())
+            .next(value => {
+              expect(value).to.be.not.null;
+            });
+        })
+          .next(() => {
+            // Add new entries, which will be added to the index.
+            return PersistencePromise.forEach(newDocPaths, (path: string) => {
+              const remoteDoc = doc(path, /*version=*/ 2, { data: 1 });
+              const dbRemoteDoc = TEST_SERIALIZER.toDbRemoteDocument(
+                remoteDoc,
+                remoteDoc.version
+              );
+              return remoteDocumentStore.put(
+                remoteDoc.key.path.toArray(),
+                dbRemoteDoc
+              );
+            });
+          })
+          .next(() => {
+            // Verify that we can get recent changes in a collection filtered by read time.
+            const docsRead: string[] = [];
+            const lastReadTime = TEST_SERIALIZER.toDbTimestampKey(version(0));
+            const range = IDBKeyRange.lowerBound(
+              ['coll2', ...lastReadTime],
```

Comment:
I think `...lastReadTime` should just be `lastReadTime`.  Please add a test to confirm and fix.

Reply:
Yes, you are correct. The proper test will use this query when it is part of the RemoteDocumentCache, but I also added a test here. I did introduce some helpers to reduce some of the code duplication this added.